cpp/include/raft/distance/detail/distance.cuh

```diff
@@ -137,7 +137,7 @@ void distance_impl(raft::resources const& handle,
   AccT* y_norm    = workspace;
   AccT* sq_x_norm = workspace;
   AccT* sq_y_norm = workspace;
-  if (x != y) {
+  if ((x != y) || ((x == y) && (m != n))) {
```

Comment:
This conditional is getting a little complicated and I see it in multiple places in this PR. Can we create a helper function that can give this a name and be reused throughout the code? It would make it a lot easier to read and maintain.

Reply:
have removed this conditional.